docs/guide/interactions/checks.rst

```diff
@@ -0,0 +1,368 @@
+:orphan:
+
+.. currentmodule:: discord
+
+.. _guide_interaction_checks:
+
+Interaction Checks Guide
+=========================
+
+Checks are a feature that allows you to set requirements in order for a command to be executed. This can include things like requiring administrator permissions, requiring a user to have a role, making sure that the bot has necessary permissions to execute a task, adding a cooldown to a command, creating a custom check function, or any combination of these.
+
+These checks will be used as decorators in conjunction with any :func:`@app_commands.command <.app_commands.command>` or :func:`@app_commands.context_menu <.app_commands.context_menu>` decorators.
+
+Before getting started, the examples shown below are made using both :class:`.app_commands.Command` and :class:`.app_commands.ContextMenu` methods. All of the checks present here can apply to either one. Let's take a look at each of the possible checks.
+
+.. note::
+
+    For information on slash commands, see the :ref:`Slash Commands Guide <discord_slash_commands>`. For more information on context menus, see the :ref:`Context Menus Guide <guide_interactions_context-menus>`.
+
+.. _guide_interaction_checks_role-check:
+
+Role Checks
+------------
+
+These checks allow a simple way to ensure that the user triggering a command has a role, or any role from a set of roles.
+
+.. code-block:: python3
+    :emphasize-lines: 2,7
+
+    # Single role check.
+    @app_commands.command()
+    @app_commands.checks.has_role('Role Name')
+    async def single_role_check(interaction: discord.Interaction):
+        await interaction.response.send_message('You have the role!', ephemeral=True)
+
+    # Any role from set of roles check.
+    @app_commands.context_menu()
+    @app_commands.checks.has_any_role('Role Name', 1234567890, 'AnotherOne')
+    async def multiple_role_check(interaction: discord.Interaction, user: discord.User):
+        await interaction.response.send_message('You have all the roles!', ephemeral=True)
```

Comment:
```suggestion
        await interaction.response.send_message('You have one of the roles!', ephemeral=True)
```
Minor thing, the response here says "you have **all** roles" but the check used is for any of the roles (ie, at least 1)